x-pack/elastic-agent/pkg/composable/providers/kubernetes/config.go

```diff
@@ -9,24 +9,53 @@ package kubernetes
 
 import (
 	"time"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
 // Config for kubernetes provider
 type Config struct {
+	Scope     string    `config:"scope"`
+	Resources Resources `config:"resources"`
+}
+
+// Resources config section for resources' config blocks
+type Resources struct {
+	Pod     *ResourceConfig `config:"pod"`
+	Node    *ResourceConfig `config:"node"`
+	Service *ResourceConfig `config:"service"`
+}
+
+// ResourceConfig for kubernetes resource
+type ResourceConfig struct {
 	KubeConfig     string        `config:"kube_config"`
+	Namespace      string        `config:"namespace"`
 	SyncPeriod     time.Duration `config:"sync_period"`
 	CleanupTimeout time.Duration `config:"cleanup_timeout" validate:"positive"`
 
 	// Needed when resource is a pod
 	Node string `config:"node"`
-
-	// Scope of the provider (cluster or node)
-	Scope string `config:"scope"`
 }
 
 // InitDefaults initializes the default values for the config.
 func (c *Config) InitDefaults() {
-	c.SyncPeriod = 10 * time.Minute
-	c.CleanupTimeout = 60 * time.Second
+	if c.Resources.Pod == nil {
+		c.Resources.Pod = &ResourceConfig{}
```

Comment:
Only Pod is enabled by default. Could node also be enabled by default, but it's only the node it's running on being by default it is scoped to `node`?

That would allow an input like below:

```
inputs:
  - type: logfile
    path: /var/log/${kubernetes.node.id}.log
```

^ Just a bad example :-)

But using that would auto add processor information for the node metadata. Being that node is priority 0, that ensures that you only want node information, versus all the pod information on those events.

Reply:
That's a good idea, I will make `node` resource to be enabled by default.